node/pkg/watchers/evm/ccq.go

```diff
@@ -757,14 +644,67 @@ func ccqBuildBatchFromCallData(req EthCallDataIntf, callBlockArg interface{}) ([
 				ecd.callTransactionArg,
 				callBlockArg,
 			},
-			Result: ecd.callResult,
+			Result: ecd.CallResult,
 			Error:  ecd.callErr,
 		})
 	}
 
 	return batch, evmCallData
 }
 
+// ccqVerifyBlockResult does basic verification on the results of the block query.
+func (w *Watcher) ccqVerifyBlockResult(blockError error, blockResult connectors.BlockMarshaller) error { //nolint:unparam
+	if blockError != nil {
+		return blockError
+	}
+
+	if blockResult.Number == nil {
+		return fmt.Errorf("block result is nil")
+	}
+
+	if blockResult.Number.ToInt().Cmp(w.ccqMaxBlockNumber) > 0 {
+		return fmt.Errorf("block number is too large")
+	}
+
+	return nil
+}
+
+// ccqVerifyAndExtractQueryResults verifies the array of call results and returns a vector of those results to be published.
+func (w *Watcher) ccqVerifyAndExtractQueryResults(requestId string, evmCallData EvmCallData) ([][]byte, error) {
+	var err error
+	results := [][]byte{}
+	for idx, evmCD := range evmCallData {
+		if evmCD.callErr != nil {
+			w.ccqLogger.Error("call failed", zap.String("requestId", requestId), zap.Int("idx", idx), zap.Error(evmCD.callErr))
+			if err != nil {
+				err = fmt.Errorf("call %d failed: %w", idx, evmCD.callErr)
```

Comment:
When you format these err's in this function, will only the last error be returned?